src/ui/SearchAlerts/SearchAlerts.ts

```diff
@@ -1,298 +1,319 @@
+        <div class='coveo-subscriptions-panel-action coveo-subscriptions-panel-action-unfollow'>${ l('SearchAlerts_unFollowing')}</div>
+        <div class='coveo-subscriptions-panel-action coveo-subscriptions-panel-action-follow'>${ l('SearchAlerts_follow')}</div>
+      </td>`;
+
+    var noSearchAlerts = container.find('.coveo-subscriptions-panel-no-subscriptions');
+
+    element.insertBefore(noSearchAlerts)
+
+    var frequencyInput = <HTMLInputElement>element.find('.coveo-subscriptions-panel-frequency select');
+
+    frequencyInput.value = subscription.frequency;
+
+    $$(frequencyInput).on('change', (event) => {
+      subscription.frequency = frequencyInput.value;
+      this.updateAndSyncSearchAlert(subscription);
+    })
+
+    $$(element.find('.coveo-subscriptions-panel-action-unfollow')).on('click', () => {
+      element.addClass('coveo-subscription-unfollowed');
       this.queryController.getEndpoint()
-        .updateSubscription(subscription)
-        .then((updated: ISubscription) => $.extend(subscription, updated))
+        .deleteSubscription(subscription)
+        .then(() => {
+          delete subscription.id;
+          var eventArgs: ISearchAlertEventArgs = { subscription: subscription };
+          $$(this.root).trigger(SearchAlertEvents.searchAlertDeleted, eventArgs);
+        })
         .catch(() => {
           this.handleSearchAlertsFail();
-        });
-    }
-
-    // FollowQuery
-    public followQuery() {
-      var queryBuilder = this.queryController.createQueryBuilder({});
-      var request = SearchAlerts.buildFollowQueryRequest(queryBuilder.build(), this.options);
+        })
+    });
 
-      this.queryController.getEndpoint().follow(request)
-        .then((subscription: ISubscription) => {
-          var eventArgs: SearchAlertEventArgs = {
-            subscription: subscription,
-            dom: this.findQueryBoxDom()
-          };
-          $(this.root).trigger(SearchAlertEvents.searchAlertCreated, eventArgs);
+    $$(element.find('.coveo-subscriptions-panel-action-follow')).on('click', () => {
+      element.removeClass('coveo-subscription-unfollowed');
+      this.queryController.getEndpoint()
+        .follow(subscription)
+        .then((updatedSearchAlert) => {
+          subscription.id = updatedSearchAlert.id;
+          var eventArgs: ISearchAlertEventArgs = { subscription: subscription };
+          $$(this.root).trigger(SearchAlertEvents.searchAlertCreated, eventArgs);
         })
         .catch(() => {
-          var eventArgs: SearchAlertsFailEventArgs = {
+          this.handleSearchAlertsFail();
+        })
+    });
+  }
+
+  private updateAndSyncSearchAlert(subscription: ISubscription) {
+    this.queryController.getEndpoint()
+      .updateSubscription(subscription)
+      .then((updated: ISubscription) => _.extend(subscription, updated))
+      .catch(() => {
+        this.handleSearchAlertsFail();
+      });
+  }
+
+  // FollowQuery

```

Comment:
Remove this comment and add the jsdoc for those component. They will need to be documented like all other component. 

This means all options and public methods.

You can also reference this https://coveord.atlassian.net/browse/JSUI-770 in your commit message.